File: src/main/scala/org/apache/flink/contrib/tensorflow/models/savedmodel/SavedModelLoader.java
package org.apache.flink.contrib.tensorflow.models.savedmodel;

import org.apache.flink.contrib.tensorflow.models.Model;
import org.tensorflow.SavedModelBundle;
import org.tensorflow.framework.MetaGraphDef;

/**
 * Abstract loader to inspect and load a saved model.
 */
public interface SavedModelLoader {

	/**
	 * Get the metagraph associated with the model.
	 * <p>
	 * <p>The metagraph is considered inexpensive and safe to load,
	 * such as during Flink program definition.
	 */
	MetaGraphDef metagraph();

	/**
	 * Load the saved model bundle.
	 */
	SavedModelBundle load();
}

File: flink-tensorflow/src/main/scala/org/apache/flink/contrib/tensorflow/models/savedmodel/SavedModelLoader.scala
package org.apache.flink.contrib.tensorflow.models.savedmodel

import org.tensorflow.SavedModelBundle
import org.tensorflow.framework.MetaGraphDef

/**
  * Abstract loader to inspect and load a saved model.
  */
trait SavedModelLoader {
  /**
    * Get the metagraph associated with the model.
    *
    * The metagraph is considered inexpensive and safe to load,
    * such as during Flink program definition.
    */
  def metagraph: MetaGraphDef

  /**
    * Load the saved model bundle.
    */
  def load(): SavedModelBundle
}
